### PR TITLE
Integrate Error Modal Functionality

### PR DESCRIPTION
**Pull request by shradhaj06**
## Description

This PR integrates a global error handling mechanism into the `ticket-tracker` application. It introduces an `ErrorProvider` context to manage error states and an `ErrorModal` component to display error messages. This setup allows for centralized error management across the application.

## Changes

1. **Context Setup:**
   - Created `ErrorContext.js` to manage error states.
   - Added `useError` hook for easier access to the error context.

2. **Error Modal Component:**
   - Created `ErrorModal.jsx` to display error messages.
   - Styled the modal using `ErrorModal.css`.

3. **Integration in App:**
   - Wrapped the application in the `ErrorProvider` context.
   - Included the `ErrorModal` component in `App.jsx`.

4. **Example Error Trigger:**
   - Added a button in the `Home` component to manually trigger an error for testing purposes.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,30 @@
+// src/App.jsx
 import React from 'react';
 import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
 import './App.css';
 import Home from './Screens/Home/Home';
 import TicketDetails from './Screens/TicketDetails/TicketDetails';
 import Sidebar from './Components/Sidebar/Sidebar';
+import { ErrorProvider } from './contexts/ErrorContext';
+import ErrorModal from './Components/ErrorModal/ErrorModal';
```
Comment (Jha2508) [inline src/App.js]: No changes should be present in `App.js` please remove once the code is tested

File: src/contexts/ErrorContext.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// src/contexts/ErrorContext.js
+import React, { createContext, useState, useContext } from 'react';
+
+const ErrorContext = createContext();
+
+export const useError = () => useContext(ErrorContext);
+
+export const ErrorProvider = ({ children }) => {
```
Comment (Jha2508) [inline src/contexts/ErrorContext.jsx:17]: docs

File: src/Screens/Home/Home.jsx [hunk 3]
```diff
@@ -1,9 +1,19 @@
+// src/Screens/Home/Home.jsx
 import React from 'react';
+import { useError } from '../../contexts/ErrorContext';
 
-function Home() {
-  return <div>Home Page
-    <p> I am the home page</p>
-  </div>;
-}
+const Home = () => {
```
Comment (Jha2508) [inline src/Screens/Home/Home.jsx]: docs

File: src/contexts/ErrorContext.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  const [error, setError] = useState(null);
+
+  const showError = (message) => {
```
Comment (Jha2508) [inline src/contexts/ErrorContext.jsx]: no need to have two functions for the purpose of changing state for `error`, formulate a single function `toggleErrorState` which can take a parameter based on which this value can be changed accordingly

File: src/App.js (hunk 1 above)
Comment (shradhaj06) [reply src/App.js]: ok

File: .idea/vcs.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (Jha2508) [inline .idea/vcs.xml]: why is directory pushed at all?

Comment (shradhaj06) [reply .idea/vcs.xml]: removed

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**